tests/events.test.ts

```diff
@@ -0,0 +1,55 @@
+import { expect } from "chai";
+import Event from "../src/classes/Event";
+import Store from "../src/classes/Store";
+import SubgraphData from "../src/classes/SubgraphData";
+import "../src/extensions/object";
+import "../src/extensions/string";
+import Gravatar from "./mocks/Gravatar";
+import yamlString from "./mocks/subgraphYml";
+
+describe("Contract events", () => {
+  const store = new Store();
+
+  class NewGravatar extends Event {
+    constructor(params?: any) {
+      super("NewGravatar", params);
+    }
+  }
+
+  const newGravatar: NewGravatar = new NewGravatar({
+    color: "purple",
+    displayName: "Harold",
+    id: "444",
+    owner: "0x1234567",
+  });
+
+  // Example mapping function
+  function handleNewGravatar(event: NewGravatar): void {
```

Comment:
I don't understand why this is part of the unit test and not the mock/tested subgraph?

Reply:
because the user defined their own mappings, I think